### PR TITLE
Moquette broker kills parent application if embedded and broker port already used #641

### PR DESCRIPTION
**Pull request by sbaranov-iqmessenger**
## Release notes
Avoid Server.startServer to exit the process it port is bound, just raise an error (#646)

[#641](https://github.com/moquette-io/moquette/issues/641)
just there was removed System.exit(1)

### CODE REVIEW

**Comment by andsel** [comment]
The `io.moquette.broker.Server` class is also used in `bat` and `sh` scripts used by the tar.gz distribution of Moquette. To complete this PR we should create another class (maybe `io.moquette.broker.Launcher` that simply wrap `Server` and catch for the exceptions, because in case of bind error, it should exit with error code without ugly traces in the console.

**Comment by sbaranov-iqmessenger** [comment]
> The `io.moquette.broker.Server` class is also used in `bat` and `sh` scripts used by the tar.gz distribution of Moquette. To complete this PR we should create another class (maybe `io.moquette.broker.Launcher` that simply wrap `Server` and catch for the exceptions, because in case of bind error, it should exit with error code without ugly traces in the console.

I didn't change current exception "culture" and don't throw BindException to not add additional exceptions in method signatures. What do you think ?

File: broker/src/main/java/io/moquette/broker/NewNettyAcceptor.java [hunk 1]
```diff
@@ -223,7 +223,7 @@ public void initChannel(SocketChannel ch) throws Exception {
         } catch (Exception ex) {
             if (ex instanceof BindException) {
                LOG.error("Cannot bind to port: " + port, ex);
-               System.exit(1);
+                throw new RuntimeException(ex);
```
Comment (andsel) [inline broker/src/main/java/io/moquette/broker/NewNettyAcceptor.java]:
```suggestion
               throw new RuntimeException("Cannot bind to port: " + port, ex);
```

**Review by andsel — CHANGES_REQUESTED** [review]
We are almost there, please integrate the suggestion and we are good to merge.

**Comment by sbaranov-iqmessenger** [comment]
> We are almost there, please integrate the suggestion and we are good to merge.

I've committed what you suggested but is there any benefit of containing human readable "Cannot bind..." because in `Server.main` I don't log exception. I just call System.exit(1) without any logging

**Comment by andsel** [comment]
> is there any benefit of containing human readable "Cannot bind..."

I think it's useful in the code that use that method, incorporating it as a library, maybe it want to log the cause of the error, if we don't throw a specific error excpetion.

**Comment by sbaranov-iqmessenger** [comment]
> > is there any benefit of containing human readable "Cannot bind..."
> 
> I think it's useful in the code that use that method, incorporating it as a library, maybe it want to log the cause of the error, if we don't throw a specific error excpetion.

oh, yeah, you are right. I got it

**Comment by andsel** [comment]
Thanks a lot @sbaranov-iqmessenger for contributing! Warm welcome

**Pull request merged**